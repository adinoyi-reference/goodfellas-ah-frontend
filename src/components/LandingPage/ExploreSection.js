import React from 'react';
import Card from '../../containers/ArticleCard';

const ExploreSection = (props) => (
  <section className="hero-lowerbody">
    <div className="hero-lowerbody-details">
      <div className="hero-lowerbody-title">
        <div className="hero-bookImage-wrapper ">
          <img src="https://res.cloudinary.com/drmmqcxkc/image/upload/v1541426068/Authors%20Haven/icons8-heart-outline-48.png" alt="" />
        </div>
        <h2>HEREâ€™S OUR BEST PICK FOR AUGUST</h2>
      </div>
      <p>
        We have buillt a community of talentrd authors who love to write lorem
        ips Lorem ipsum dolor sit amet consectetur, adipisicing elit.
        Cupiditate Lorem ipsum dolor sit amet consectetur, adipisicing Lorem
        ipsum dolor sit amet consectetur, adipisicing elit. Cupiditatelit.
        Cupiditate
      </p>
    </div>
    <div className="hero-lowerbody-articles">
      <Card {...props} />
    </div>
  </section>
);

export default ExploreSection;
