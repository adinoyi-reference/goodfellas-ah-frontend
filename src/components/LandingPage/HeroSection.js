import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import Header from '../shared/Header';
import Button from '../shared/Button';

const Hero = (props) => {
  const { auth } = props;
  console.log(auth);
  return (
    <section className="hero-section">
      <div className="hero-section-overlay">
        <Header parentComponent="landingpage" />
        <h2>Write it out beautifully</h2>
        <div className="hero-section-description">
          <p>
            Or type it, its just as fine. Here, we help get your articles across
          </p>
          <p>to a pool of eager readers. We know there are no limits to your</p>
          <p>imagination. We canâ€™t wait to meet you.</p>
        </div>
        {auth
          ? (
            <div className="hero-section-buttons">
              <Link to="/articles/create"><Button className="btn hero-section-greenbutton" type="button" title="CREATE ARTICLES" /></Link>
              <Link to="/articles/home"><Button className="btn hero-section-whitebutton" type="button" title="READ ARTICLES" /></Link>
            </div>
          )
          : (
            <div className="hero-section-buttons">
              <Link to="/auth/signin"><Button className="btn hero-section-greenbutton" type="button" title="CREATE ARTICLES" /></Link>
              <Link to="/auth/signin"><Button className="btn hero-section-whitebutton" type="button" title="READ ARTICLES" /></Link>
            </div>
          )
        }
      </div>
      <div className="hero-body-vector-white" />
      <div>
        <div className="big-hero-card row">
          <div className="hero-card-details col-sm-4">
            <h6>Why a willing kid</h6>
            <h6>would win an adult</h6>
            <h6>in a fist fight.</h6>
            <p>
              This is my unique story <br /> of how a savage kid from
              <br /> the west devotedly kicked
              <br /> a white glass.....
            </p>
            <div className="row hero-card-author">
              <img
                src="https://res.cloudinary.com/drmmqcxkc/image/upload/v1541426069/Authors%20Haven/john.jpg"
                alt="Author Profile"
              />
              <p className="col-sm-8">John Adewale</p>
            </div>
          </div>
          <div className="hero-card-image col-sm-6" />
        </div>
      </div>
    </section>
  );
};

function mapStatetoProps(state) {
  return { auth: state.auth.authenticated };
}

export default connect(mapStatetoProps, null)(Hero);
