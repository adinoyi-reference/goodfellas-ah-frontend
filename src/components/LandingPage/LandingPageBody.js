import React from 'react';
import { Link } from 'react-router-dom';

const LandingPageBody = () => (
  <section className="hero-body-background">
    <div className="hero-body">
      <div className="hero-body-details">
        <div className="hero-body-title row">
          <h2 className="col-lg-9">
            THE COMPLETE DIGITAL TOOL FOR AUTHORS WILLING TO SHARE
          </h2>
          <div className="hero-bookImage-wrapper ">
            <img
              src="https://res.cloudinary.com/drmmqcxkc/image/upload/v1541426068/Authors%20Haven/icons8-literature-48.png"
              alt=""
            />
          </div>
        </div>
        <p>
          Authors Haven taps into a community of like-minded authors to
          foster inspiration in you. Enabling the world’s most insightful
          writers, thinkers, and storytellers to bring you the smartest takes on
          topics that matter. So whatever your interest, you can always find
          fresh thinking and unique perspectives here.
        </p>
        <hr />
        <Link className="hero-signup" to="/auth/signup">
          <p>Writer? We can’t wait to meet you &rarr;</p>
        </Link>
      </div>
    </div>
    <div className="hero-body-vector-blue" />
  </section>
);

export default LandingPageBody;
