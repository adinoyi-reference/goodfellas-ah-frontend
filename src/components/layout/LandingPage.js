import React from 'react';
import Footer from '../shared/Footer';
import Hero from '../LandingPage/HeroSection';
import Body from '../LandingPage/LandingPageBody';
import Explore from '../LandingPage/ExploreSection';

const LandingPageLayout = (props) => (
  <div>
    <Hero {...props} />
    <Body />
    <Explore />
    <Footer />
  </div>
);

export default LandingPageLayout;
