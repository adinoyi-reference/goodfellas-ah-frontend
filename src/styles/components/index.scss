// Colours
$maingreen: #57B182;
$faintblue: #F4F9FC;

:root {
  --light-gray: #a6a2a2;
  --lighter-gray: rgba(36, 32, 56, 0.17);
}
body {
  font-family: 'Roboto', sans-serif;
}

.button {
  padding: 5px 20px;
  min-width: 7rem;
  text-align: center;
  cursor: pointer;
  border-radius: 5px;
  transition: all .3s ease-in;
  font-size: 14px;
  &:hover {
    color: white !important;
    border-color: transparent !important;
    background: var(--light-gray);
  }
  &:focus {
    outline: none;
  }
}
h3.username {
  font-size: 24px;
  font-weight: bold;
  text-transform: capitalize;
}
.outline {
  color: var(--light-gray);
  border: 1.2px solid var(--light-gray);
  border-radius: 0;
  background-color: transparent;
  &.disabled {
    border: 1.2px solid var(--light-gray) !important;
    color: var(--light-gray) !important;
    &:hover {
      cursor: default;
      background: transparent;
    }

  }
}
a.outline:hover {
  text-decoration: none;
}
.hoverable {
  transition: all .2s ease-in-out;
  &:hover {
    transform: translateY(-1px);
    box-shadow: 1px 5px 15px rgba(0, 0, 0, 0.2);
  }
}
.icon {
  display: inline-block;
  width: 1em;
  height: 1em;
  stroke-width: 0;
  stroke: currentColor;
  fill: currentColor;
}
.centralizer {
  display: flex;
  justify-content: center;
  align-items: center;
}
.no-record {
  min-height: 300px;
  font-size: 25px;
  font-weight: 100;
  text-align: center;
  color: var(--lighter-gray);
  .icon {
    flex: 0 0 25px;
  }
}
.profile-image {
  border: 1px solid var(--lighter-gray);
  height: 5rem;
  width: 5rem;
  border-radius: 50%;
  background-image: url(https://res.cloudinary.com/drmmqcxkc/image/upload/v1541506955/user-placeholder.png);
  background-size: 100%;
  @media screen and (min-width: 500px) {
    height: 7rem;
    width: 7rem;    
  }
  font-family: 'Roboto', sans-serif;
}

.extra-buttons {
  max-height: 0;
  height: 100%;
  overflow-y: hidden;
  opacity: .1;
  margin-bottom: -20px;
  transition: all .3s ease-in-out;
  &.active {
    margin-top: 5px;
    max-height: 400px;
    opacity: 1;
    padding-top: 10px;
  }
}

.extra-buttons img{
  width: 40px;
  height: auto;
  margin-left: 10px;
  margin-bottom: 17px;
}