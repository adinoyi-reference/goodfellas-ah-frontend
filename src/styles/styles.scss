@import '../../node_modules/bootstrap/scss/bootstrap.scss';
<<<<<<< HEAD
@import 'components/index.scss';
@import 'components/pwreset.scss';
@import 'components/header.scss';
@import 'components/loading.scss';
@import 'components/footer.scss';

=======
@import './components/fonts.scss';
@import './components/auth.scss';
@import './components/auth-form.scss';
@import './components/author-writing.scss';
>>>>>>> staging
