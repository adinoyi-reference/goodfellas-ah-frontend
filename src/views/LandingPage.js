import React from 'react';
import LandingPageLayout from '../components/layout/LandingPage';

const LandingPage = () => (
  <div>
    <LandingPageLayout />
  </div>
);

export default (LandingPage);
