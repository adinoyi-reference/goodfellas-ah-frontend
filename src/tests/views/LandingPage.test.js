import React from 'react';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import Root from '../../root';
import LandingPage from '../../views/LandingPage';
import '../../styles/styles.scss';

let wrapped;
const mockState = {
  auth: {
    authenticated: null,
    userId: null,
    user: null,
    ownProfile: { userId: 1 },
  },
};
beforeEach(() => {
  wrapped = mount(
    <Root initialState={mockState}>
      <MemoryRouter initialEntries={[{ key: 'testkey' }]}>
        <LandingPage />
      </MemoryRouter>
    </Root>,
  );
});

afterEach(() => wrapped.unmount());

describe('Landing Page UI', () => {
  describe('render features', () => {
    test('container should render as expected', () => {
      const tree = toJson(wrapped);
      expect(tree).toMatchSnapshot();
    });
  });
});
