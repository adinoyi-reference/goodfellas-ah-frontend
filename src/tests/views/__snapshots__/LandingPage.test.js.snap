// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page UI render features container should render as expected 1`] = `
<_default
  initialState={
    Object {
      "auth": Object {
        "authenticated": null,
        "ownProfile": Object {
          "userId": 1,
        },
        "user": null,
        "userId": null,
      },
    }
  }
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <MemoryRouter
      initialEntries={
        Array [
          Object {
            "key": "testkey",
          },
        ]
      }
    >
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "key": "testkey",
                "pathname": "/",
                "search": "",
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "key": "testkey",
              "pathname": "/",
              "search": "",
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <LandingPage>
          <div>
            <LandingPageLayout>
              <div>
                <Connect(Hero)>
                  <Hero
                    auth={null}
                    dispatch={[Function]}
                  >
                    <section
                      className="hero-section"
                    >
                      <div
                        className="hero-section-overlay"
                      >
                        <Connect(Header)
                          auth={null}
                          dispatch={[Function]}
                          parentComponent="landingpage"
                        >
                          <Header
                            auth={null}
                            dispatch={[Function]}
                            getNotification={[Function]}
                            inAppStatus={
                              Object {
                                "articleLoading": false,
                                "articles": Array [],
                                "authenticated": "",
                                "error": Array [],
                                "errorMessage": "",
                                "notification": Array [],
                                "notifications": Object {},
                                "ownProfile": Object {},
                                "profile": Object {
                                  "articles": Array [],
                                  "favorites": Array [],
                                  "followers": Array [],
                                  "following": Array [],
                                  "loading": true,
                                  "profile": Object {},
                                  "profileError": "",
                                  "profileView": "Following",
                                  "user": Object {},
                                },
                                "searchError": Array [],
                                "searchResults": Array [],
                                "successMessage": "",
                                "tags": Array [],
                                "tagsError": Array [],
                                "user": Object {},
                              }
                            }
                            notifications={Object {}}
                            parentComponent="landingpage"
                            profile={
                              Object {
                                "userId": 1,
                              }
                            }
                            profileNavigation={[Function]}
                            search={[Function]}
                            searchError={Array []}
                            searchResults={Array []}
                            seenNotification={[Function]}
                            setNotification={[Function]}
                            signout={[Function]}
                          >
                            <div>
                              <header
                                className="landing-page-header"
                              >
                                <nav
                                  className="navbar"
                                >
                                  <Link
                                    className="navbar-title navbar-brand"
                                    replace={false}
                                    to="/"
                                  >
                                    <a
                                      className="navbar-title navbar-brand"
                                      href="/"
                                      onClick={[Function]}
                                    >
                                      <h3
                                        className="header-white"
                                      >
                                        Authors Haven
                                      </h3>
                                    </a>
                                  </Link>
                                  <div
                                    className="navbar-unauthorized"
                                  >
                                    <div
                                      className="navbar-main-wrapper"
                                    >
                                      <div>
                                        <form
                                          onSubmit={[Function]}
                                        >
                                          <div
                                            className="header-user-search-unauth"
                                          >
                                            <input
                                              className="searchbar"
                                              id="Title"
                                              onChange={[Function]}
                                              placeholder="Search for articles"
                                              type="search"
                                            />
                                            <span
                                              className="search-click"
                                              onClick={[Function]}
                                            >
                                              <img
                                                alt=""
                                                className="search-dropdown"
                                                src="https://res.cloudinary.com/drmmqcxkc/image/upload/v1542628198/Authors%20Haven/dropdown-arrow-white.png"
                                              />
                                            </span>
                                            <span />
                                            <span
                                              className="searchbar-toggle"
                                              onClick={[Function]}
                                            >
                                              <img
                                                alt=""
                                                src="https://res.cloudinary.com/drmmqcxkc/image/upload/v1542628197/Authors%20Haven/search-icon-white.svg"
                                              />
                                            </span>
                                            <span
                                              className="searchbar-toggle-mobile"
                                              onClick={[Function]}
                                            >
                                              <img
                                                alt=""
                                                src="https://res.cloudinary.com/drmmqcxkc/image/upload/v1542628197/Authors%20Haven/search-icon-white.svg"
                                              />
                                            </span>
                                            <div
                                              className="search-parameters"
                                            >
                                              Filter by
                                              <select>
                                                <option>
                                                  Author
                                                </option>
                                                <option>
                                                  Tag
                                                </option>
                                              </select>
                                              <input
                                                id="searchKeyword"
                                                onChange={[Function]}
                                                placeholder="Enter keyword"
                                                type="text"
                                              />
                                              <Button
                                                className="btn hero-section-greenbutton"
                                                title="Search"
                                                type="Submit"
                                              >
                                                <button
                                                  className="btn hero-section-greenbutton"
                                                  type="Submit"
                                                >
                                                  Search
                                                </button>
                                              </Button>
                                            </div>
                                          </div>
                                        </form>
                                      </div>
                                      <button
                                        className="navbar-toggler"
                                        data-target="#navbarNav"
                                        data-toggle="collapse"
                                        onClick={[Function]}
                                        type="button"
                                      >
                                        <span
                                          className=""
                                        >
                                          <img
                                            alt=""
                                            className="menuIcon"
                                            src="https://res.cloudinary.com/drmmqcxkc/image/upload/v1541426069/Authors%20Haven/icons8-menu.svg"
                                          />
                                        </span>
                                      </button>
                                      <div
                                        className="navbar-wrapper"
                                      >
                                        <ul
                                          className="navbar-element"
                                        >
                                          <Link
                                            className="nav-item nav-link"
                                            replace={false}
                                            to="/auth/signin"
                                          >
                                            <a
                                              className="nav-item nav-link"
                                              href="/auth/signin"
                                              onClick={[Function]}
                                            >
                                              Sign In
                                            </a>
                                          </Link>
                                          <p
                                            className="link-border"
                                          >
                                            |
                                          </p>
                                          <Link
                                            className=" nav-item nav-link"
                                            replace={false}
                                            to="/auth/signup"
                                          >
                                            <a
                                              className=" nav-item nav-link"
                                              href="/auth/signup"
                                              onClick={[Function]}
                                            >
                                              Sign Up
                                            </a>
                                          </Link>
                                        </ul>
                                      </div>
                                    </div>
                                  </div>
                                </nav>
                              </header>
                            </div>
                          </Header>
                        </Connect(Header)>
                        <h2>
                          Write it out beautifully
                        </h2>
                        <div
                          className="hero-section-description"
                        >
                          <p>
                            Or type it, its just as fine. Here, we help get your articles across
                          </p>
                          <p>
                            to a pool of eager readers. We know there are no limits to your
                          </p>
                          <p>
                            imagination. We can’t wait to meet you.
                          </p>
                        </div>
                        <div
                          className="hero-section-buttons"
                        >
                          <Link
                            replace={false}
                            to="/auth/signin"
                          >
                            <a
                              href="/auth/signin"
                              onClick={[Function]}
                            >
                              <Button
                                className="btn hero-section-greenbutton"
                                title="CREATE ARTICLES"
                                type="button"
                              >
                                <button
                                  className="btn hero-section-greenbutton"
                                  type="button"
                                >
                                  CREATE ARTICLES
                                </button>
                              </Button>
                            </a>
                          </Link>
                          <Link
                            replace={false}
                            to="/auth/signin"
                          >
                            <a
                              href="/auth/signin"
                              onClick={[Function]}
                            >
                              <Button
                                className="btn hero-section-whitebutton"
                                title="READ ARTICLES"
                                type="button"
                              >
                                <button
                                  className="btn hero-section-whitebutton"
                                  type="button"
                                >
                                  READ ARTICLES
                                </button>
                              </Button>
                            </a>
                          </Link>
                        </div>
                      </div>
                      <div
                        className="hero-body-vector-white"
                      />
                      <div>
                        <div
                          className="big-hero-card row"
                        >
                          <div
                            className="hero-card-details col-sm-4"
                          >
                            <h6>
                              Why a willing kid
                            </h6>
                            <h6>
                              would win an adult
                            </h6>
                            <h6>
                              in a fist fight.
                            </h6>
                            <p>
                              This is my unique story 
                              <br />
                               of how a savage kid from
                              <br />
                               the west devotedly kicked
                              <br />
                               a white glass.....
                            </p>
                            <div
                              className="row hero-card-author"
                            >
                              <img
                                alt="Author Profile"
                                src="https://res.cloudinary.com/drmmqcxkc/image/upload/v1541426069/Authors%20Haven/john.jpg"
                              />
                              <p
                                className="col-sm-8"
                              >
                                John Adewale
                              </p>
                            </div>
                          </div>
                          <div
                            className="hero-card-image col-sm-6"
                          />
                        </div>
                      </div>
                    </section>
                  </Hero>
                </Connect(Hero)>
                <LandingPageBody>
                  <section
                    className="hero-body-background"
                  >
                    <div
                      className="hero-body"
                    >
                      <div
                        className="hero-body-details"
                      >
                        <div
                          className="hero-body-title row"
                        >
                          <h2
                            className="col-lg-9"
                          >
                            THE COMPLETE DIGITAL TOOL FOR AUTHORS WILLING TO SHARE
                          </h2>
                          <div
                            className="hero-bookImage-wrapper "
                          >
                            <img
                              alt=""
                              src="https://res.cloudinary.com/drmmqcxkc/image/upload/v1541426068/Authors%20Haven/icons8-literature-48.png"
                            />
                          </div>
                        </div>
                        <p>
                          Authors Haven taps into a community of like-minded authors to foster inspiration in you. Enabling the world’s most insightful writers, thinkers, and storytellers to bring you the smartest takes on topics that matter. So whatever your interest, you can always find fresh thinking and unique perspectives here.
                        </p>
                        <hr />
                        <Link
                          className="hero-signup"
                          replace={false}
                          to="/auth/signup"
                        >
                          <a
                            className="hero-signup"
                            href="/auth/signup"
                            onClick={[Function]}
                          >
                            <p>
                              Writer? We can’t wait to meet you →
                            </p>
                          </a>
                        </Link>
                      </div>
                    </div>
                    <div
                      className="hero-body-vector-blue"
                    />
                  </section>
                </LandingPageBody>
                <ExploreSection>
                  <section
                    className="hero-lowerbody"
                  >
                    <div
                      className="hero-lowerbody-details"
                    >
                      <div
                        className="hero-lowerbody-title"
                      >
                        <div
                          className="hero-bookImage-wrapper "
                        >
                          <img
                            alt=""
                            src="https://res.cloudinary.com/drmmqcxkc/image/upload/v1541426068/Authors%20Haven/icons8-heart-outline-48.png"
                          />
                        </div>
                        <h2>
                          HERE’S OUR BEST PICK FOR AUGUST
                        </h2>
                      </div>
                      <p>
                        We have buillt a community of talentrd authors who love to write lorem ips Lorem ipsum dolor sit amet consectetur, adipisicing elit. Cupiditate Lorem ipsum dolor sit amet consectetur, adipisicing Lorem ipsum dolor sit amet consectetur, adipisicing elit. Cupiditatelit. Cupiditate
                      </p>
                    </div>
                    <div
                      className="hero-lowerbody-articles"
                    >
                      <Connect(Card)>
                        <Card
                          getArticles={[Function]}
                        >
                          <div>
                            <Loading>
                              <div
                                className="loading-icon"
                              >
                                <img
                                  alt=""
                                  src="https://res.cloudinary.com/drmmqcxkc/image/upload/v1541426069/Authors%20Haven/Spinner-1s.gif"
                                />
                              </div>
                            </Loading>
                          </div>
                        </Card>
                      </Connect(Card)>
                    </div>
                  </section>
                </ExploreSection>
                <Footer>
                  <footer>
                    <div
                      className="copyright"
                    >
                      <p>
                        ©Goodfellas
                      </p>
                    </div>
                  </footer>
                </Footer>
              </div>
            </LandingPageLayout>
          </div>
        </LandingPage>
      </Router>
    </MemoryRouter>
  </Provider>
</_default>
`;
